helix-loader/src/grammar.rs

```diff
@@ -471,6 +472,7 @@ fn build_tree_sitter_library(
             .arg(header_path)
             .arg("-o")
             .arg(&library_path)
+            .arg("-std=c++14")
```

Comment:
Will the `.arg("-std=c99")` below conflict with this? Should that be removed?

Reply:
For clang and msvbc the C standard and C++ standard can be set separately when using a c++ compiler.  This makes sense since these two are really decoupled but it seems some versions of GCC are not as flexible.

 I amended to commit to always use c11 for c files and c++14 for c++ files keeping the existing file based detection in tact (and setting the appropriate standard) to (both on Windows and UNIX) to again be consistent across various toolchains and just moved the whole thing out of the if. The special handling for file extensions is actually not possible compilers should do that automatically but I left it in for now.